The fixes to time.hpp are applied to upstream SVN post 1.36.0
see http://svn.boost.org/trac/boost/changeset/47957

The fix for topological_sort.hpp reported upstream
as http://svn.boost.org/trac/boost/ticket/2209
- marked as closed in r49000

--- boost-posix-2008-03-21.orig/boost/date_time/time.hpp
+++ boost-posix-2008-03-21/boost/date_time/time.hpp
@@ -77,14 +77,14 @@
     /*! Optional bool parameter will return time zone as an offset 
      * (ie "+07:00"). Empty string is returned for classes that do 
      * not use a time_zone */
-    std::string zone_name(bool as_offset=false) const
+    std::string zone_name(bool = false) const
     {
       return time_system::zone_name(time_);
     }
     /*! Optional bool parameter will return time zone as an offset 
      * (ie "+07:00"). Empty string is returned for classes that do 
      * not use a time_zone */
-    std::string zone_abbrev(bool as_offset=false) const
+    std::string zone_abbrev(bool = false) const
     {
       return time_system::zone_name(time_);
     }
--- boost-posix-2008-03-21.orig/boost/graph/topological_sort.hpp
+++ boost-posix-2008-03-21/boost/graph/topological_sort.hpp
@@ -37,7 +37,7 @@
       : m_iter(_iter) { }
     
     template <typename Edge, typename Graph>
-    void back_edge(const Edge& u, Graph&) { throw not_a_dag(); }
+    void back_edge(const Edge&, Graph&) { throw not_a_dag(); }
     
     template <typename Vertex, typename Graph> 
     void finish_vertex(const Vertex& u, Graph&) { *m_iter++ = u; }
